chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1129,74 +1008,72 @@ impl PeerManagerActor {
     ///    until safe set has safe_set_size elements.
     fn maybe_stop_active_connection(&self) {
         let connected_peers = self.state.connected_peers.read();
-        let filter_peers = |predicate: &dyn Fn(&ConnectedPeer) -> bool| {
+        let filter_peers = |predicate: &dyn Fn(&ConnectedPeer) -> bool| -> Vec<_> {
             connected_peers
-                .iter()
-                .filter_map(|(id, peer)| predicate(peer).then(|| id))
-                .collect::<Vec<_>>()
+                .values()
+                .filter(|peer| predicate(&*peer))
+                .map(|peer| peer.peer_info.id.clone())
+                .collect()
         };
 
         // Build safe set
         let mut safe_set = HashSet::new();
 
         // If there is not enough non-whitelisted peers, return without disconnecting anyone.
-        let whitelisted_peers =
-            filter_peers(&|p| self.is_peer_whitelisted(&p.full_peer_info.peer_info));
+        let whitelisted_peers = filter_peers(&|p| self.is_peer_whitelisted(&p.peer_info));
         if connected_peers.len() - whitelisted_peers.len()
             <= self.config.ideal_connections_hi as usize
         {
             return;
         }
         // Add whitelisted nodes to the safe set.
-        safe_set.extend(whitelisted_peers.into_iter().cloned());
+        safe_set.extend(whitelisted_peers.into_iter());
```

Comment:
```suggestion
        safe_set.extend(whitelisted_peers);
```

API usually use `IntoIterator` rathre than `Iterator`, so `.into_iter()` is on the callee-side